### PR TITLE
Fixes #611

### LINKED ISSUE #611: Benchmark segfault (NvFuserScheduler_TIMM_BatchNorm_nhwc_fp16/128/72/7)
```
-- LOG(2): Running NvFuserScheduler_TIMM_BatchNorm_nhwc_fp16___GRAPH/NvFuserScheduler_TIMM_BatchNorm_nhwc_fp16/128/72/7/manual_time for 1
zsh: abort (core dumped)   ./bin/nvfuser_bench  --benchmark_min_time=0.00001
```

### PR DESCRIPTION
**Pull request by naoyam**

### CODE REVIEW

File: test/test_dynamic_type.cpp [hunk 1]
```diff
@@ -358,6 +358,8 @@ TEST_F(DynamicTypeTest, MoveCtor) {
   // This should not compile:
   // NonCopyableType bb(a);
   NonCopyableType b(std::move(a));
+  // Suppress unused var warning
+  (void)b;
```
Comment (naoyam) [inline test/test_dynamic_type.cpp:362]: Unrelated but clang fails to compile without this

File: csrc/device_lower/pass/vectorize_welford.cpp [hunk 2]
```diff
@@ -526,7 +526,7 @@ class WelfordVectorizer : public kir::ExprMutator {
 
       // Check if the predicate of the expr is known to be true
       if (expr_ite->predicate()->isConst() &&
-          expr_ite->predicate()->value()->value()) {
+          expr_ite->predicate()->value()->value().as<bool>()) {
```
Comment (naoyam) [inline csrc/device_lower/pass/vectorize_welford.cpp:529]: Fails without this change with this error:

```
  * frame #0: 0x00007fff6fcad610 libstdc++.so.6`__cxxabiv1::__cxa_throw(obj=0x00005555d9f90830, tinfo=0x00007ffff4b97560, dest=(libc10.so`c10::Error::~Error() at Exception.h:29))(void *)) at eh_throw.cc:77:1
    frame #1: 0x00007fffb1d510ba libc10.so`c10::detail::torchCheckFail(func="operator&&", file="/home/nmaruyama/nvdl/nvfuser/debug1/csrc/dynamic_type.h", line=594, msg=error: summary string parsing error) at Exception.cpp:87:3
    frame #2: 0x00007ffff7b9808e libnvfuser_codegen.so`std::enable_if<((is_dynamic_type_v<nvfuser::DynamicType<nvfuser::Containers<std::vector, nvfuser::Struct>, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool> >) && (!(is_dynamic_type_v<bool>))) && (any_check(land_ldefined_checker<bool, nvfuser::DynamicType<nvfuser::Containers<std::vector, nvfuser::Struct>, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool>::VariantType>, nvfuser::DynamicType<nvfuser::Containers<std::vector, nvfuser::Struct>, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool>::type_identities_as_tuple)), nvfuser::DynamicType<nvfuser::Containers<std::vector, nvfuser::Struct>, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool> >::type nvfuser::operator&&<bool, nvfuser::DynamicType<nvfuser::Containers<std::vector, nvfuser::Struct>, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool> >(x=0x00007fffffff562f, y=0x00005555ddaad140) at dynamic_type.h:594:1
```

Is there a better fix?

**Review by zasdfgbnm — APPROVED** [review]
This fix makes sense. The problem is, in `expr_ite->predicate()->isConst() && expr_ite->predicate()->value()->value()`, we are using the `&&` operator of `DynamicType`. However, in C++, overloading `&&` loses short-circuit behavior (see "Restrictions" in https://en.cppreference.com/w/cpp/language/operators), so for the case where the second operand of `&&` does not have a value, we will be doing `bool && std::monostate`, which does not exist. As a result, we will throw an error from `operator&&`. But unfortunately, for some reason, the `~c10::Error` triggers a segfault (I think this is not related to nvFuser.)

**Comment by zasdfgbnm** [comment]
I will think about how to implement the short-circuit behavior properly. But for now, let's just take this PR in.

**Comment by zasdfgbnm** [comment]
Oh, no, it is impossible to implement short-circuit behavior. For example:
```C++
false && (PolymorphicValue(std::monostate{}) + 1)
```
will always error out, regardless how we implement `&&`.

**Pull request merged**